svc/svc.go

```diff
@@ -111,37 +110,46 @@ func NewStandardHandler(opts HandlerOpts) http.Handler {
 // RunServer handles the biolerplate of starting an http server and handling
 // signals gracefully.
 func RunServer(h http.Handler, port string, writeTimeout time.Duration) {
-	errCh := make(chan error)
-
-	// Handle SIGINT and SIGTERM.
-	sigCh := make(chan os.Signal)
-	signal.Notify(sigCh, syscall.SIGINT, syscall.SIGTERM)
-
-	fmt.Printf("Listening on port %s\n", port)
-
 	// Add timeouts to the server
 	srv := &http.Server{
 		WriteTimeout: writeTimeout * time.Second,
 		Addr:         ":" + port,
 		Handler:      h,
 	}
 
+	fmt.Printf("HTTP server listening on port %s\n", port)
+	runServer(srv)
+}
+
+func runServer(srv *http.Server) {
+	idleConnsClosed := make(chan struct{})
+
 	go func() {
-		defer reporter.Monitor(context.Background())
-		err := srv.ListenAndServe()
-		if err != nil {
-			errCh <- errors.Wrapf(err, "unable to start server")
+		// Handle SIGINT and SIGTERM.
+		sigCh := make(chan os.Signal, 1)
```

Comment:
what did you intend to do by adding a bufferSize here as 1?

Reply:
it seems to me that the reason is that we send and receive the value on that channel in the _same_ goroutine. If `sigCh` was unbuffered, the goroutine would be blocked in `signal.Notify` trying to send a value to that channel. Whereas with a buffered channel, it sends the value and moves on, and we then block on the next line which receives a value from the channel.

See https://golang.org/doc/effective_go.html#channels.

Reply:
Almost right. From the documentation:
https://godoc.org/os/signal#Notify

```
Package signal will not block sending to c: the caller must ensure that c has sufficient buffer space to keep up with the expected signal rate. For a channel used for notification of just one signal value, a buffer of size 1 is sufficient.
```

And example:
``` golang
// Set up channel on which to send signal notifications.
// We must use a buffered channel or risk missing the signal
// if we're not ready to receive when the signal is sent.
c := make(chan os.Signal, 1)
signal.Notify(c, os.Interrupt)

// Block until a signal is received.
s := <-c
fmt.Println("Got signal:", s)
```

Reply:
So it handles a pretty minor race condition